api_client/python/timesketch_api_client/client.py

```diff
@@ -549,3 +549,24 @@ def get_sigma_rule(self, rule_uuid):
             Instance of a Sigma object.
         """
         return sigma.Sigma(rule_uuid, api=self)
+
+    def get_sigma_rule_by_text(self, rule_text):
+        """Returns a Sigma Object based on a text POST.
+
+        Args:
+            rule_text: Full Sigma rule text.
+
+        Returns:
+            Instance of a Sigma object.
+        """
+        if not rule_text:
+            raise ValueError('Rule text given.')
+
+        try:
+            sigma_rule = None #sigma_util.get_sigma_rule_by_text(rule_text)
```

Comment:
why not

```
sigma_obj = sigma.Sigma(api=self)

sigma_obj.from_text(rule_text)
```

And also with that change the `__init__` to not take in rule_uuid but rather have

```
sigma_obj = sigma.Sigma(api=self)
sigma_obj.from_rule(rule_uuid)
```

that way you can use the same mechanism to accommodate all use cases

Reply:
one of the reasons is that an API client cannot import sigma_util, the library, since that is not part of the API client codebase but rather part of timesketch, and we don't want to depend on that.

in `from_text` or `from_rule_text` you can then do some textual verifications, or other verifications and then send an API call that does the rest....

Reply:
That is a very good and valid point, will surely do that!